SteamBot/Logging/Log.cs

```diff
@@ -95,10 +99,22 @@ public void RemoveLoggingObject(LoggerBase logger)
 
         public void Dispose()
         {
-            Disposed = true;
-            foreach (IDisposable logger in LoggerObjects.OfType<IDisposable>())
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        public void Dispose(bool disposing)
+        {
+            if (!Disposed)
             {
-                logger.Dispose();
+                if (disposing)
+                {
+                    foreach (IDisposable disposableLogger in LoggerObjects.OfType<IDisposable>())
+                        disposableLogger.Dispose();
```

Comment:
An unmanaged resource is anything the garbage collector can't deal with disposing, like files, network sockets, video buffers, etc.  In .Net, classes that deal with unmanaged resources will implement `IDisposable` so you can dispose of them manually.  They will typically also have destructors to avoid permanent leaks when `Dispose()` is not called *(due to a bug in the calling problem)*, but these should not be relied on because when they are called is non-deterministic.

So, calling `Dispose` on any `IDisposable`s is the correct behavior.  You shouldn't be calling `GC.SuppressFinalize()` though.  I can't think of an instance outside of debugging where you'd **ever** want to call that.